pydm/utilities/stylesheet.py

```diff
@@ -0,0 +1,128 @@
+        timer.timeout.connect(partial(_set_style_data, style_data, timer))
+        timer.start()
+    else:
+        # For running PyDM Launcher
+        _set_style_data(style_data, None)
+
+    return style_data
+
+
+def _get_style_data(stylesheet_file_path):
+    """
+    Read the global stylesheet file and provide the style data as a str.
+
+    Parameters
+    ----------
+    stylesheet_file_path : str
+        The path to the global stylesheet.
+
+    Returns
+    -------
+    The style data read from the stylesheet file : str
+    """
+    style_data = None
+    if stylesheet_file_path is not None:
+        try:
+            with open(stylesheet_file_path, 'r') as stylesheet_file:
+                logger.info("Opening style file '{0}'...".format(stylesheet_file_path))
+                style_data = stylesheet_file.read()
+        except Exception as ex:
+            logger.error("Error reading the stylesheet file '{0}'. Exception: {1}".format(stylesheet_file_path,
+                                                                                          str(ex)))
+    else:
+        try:
+            with open(GLOBAL_STYLESHEET) as default_stylesheet:
+                logger.info("Opening the default stylesheet '{0}'...".format(GLOBAL_STYLESHEET))
+                style_data = default_stylesheet.read()
+        except Exception as ex:
+            logger.error("Cannot find the default stylesheet file '{0}'. Exception: {1}".format(GLOBAL_STYLESHEET,
+                                                                                                str(ex)))
+    return style_data
+
+
+def _set_style_data(style_data, timer):
+    """
+    Apply the global stylesheet data to a .ui form.
+
+    If the PyDM Launcher opens this .ui form, apply the stylesheet directly to the app. Otherwise, in Qt Designer, apply
+    the stylesheet to the root object ("formContainer").
+
+    Parameters
+    ----------
+    style_data : str
+        The style data obtained from a stylesheet file
+    timer : QTimer
+        The timer to launch the style data applying method (for the Qt Designer environment only)
+    """
+    app = QApplication.instance()
+    if is_pydm_app():
+        _set_stylesheet_for_app(app, style_data)
+    else:
+        for w in app.allWidgets():
+            try:
+                name = w.objectName()
+            except:
+                name = ''
+            # Set the stylesheet to the root form object
+            if name == "formContainer":
+                # This will be loaded from the file set by users to define the global stylesheet for the facility in
+                # use
+                w.setStyleSheet(style_data)
+                # A timer is fully expected here. If a timer cannot be provided for whatever reason, we must re-evaluate
+                # the current Qt Designer stylesheet application methodology
+                timer.stop()
+
+
+def _set_stylesheet_for_app(app, style_data):
```

Comment:
Do we need a method for just one line of code?

Reply:
This was needed for the unit test (https://github.com/slaclab/pydm/pull/361/files#diff-060209a2443a2f8baa84f18aefca4e77R171)